src/main/java/com/elixr/poc/service/PurchaseService.java

```diff
@@ -0,0 +1,44 @@
+package com.elixr.poc.service;
+import com.elixr.poc.data.Purchase;
+import com.elixr.poc.repository.PurchaseRepository;
+import com.elixr.poc.rest.request.PurchaseRequest;
+import com.elixr.poc.rest.response.PurchaseResponse;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+/**
+ * service class for editing response and save or update new purchase to database
+ */
+
+@Service
+public class PurchaseService {
+    private final PurchaseRepository purchaseRepository;
+
+    public PurchaseService(PurchaseRepository purchaseRepository) {
+        this.purchaseRepository = purchaseRepository;
+    }
+
+    private static Purchase createPurchaseObjectFromRequest(PurchaseRequest purchaseRequest) {
+        return Purchase.builder().userName(purchaseRequest.getUserName()).product(purchaseRequest.getProduct()).amount(purchaseRequest.getAmount()).date(purchaseRequest.getDate()).build();
+    }
+
+    public PurchaseResponse createValidPurchase(PurchaseRequest purchaseRequestObject) {
```

Comment:
Are you creating another method with name createInvalidPurchase()? 
If not just createPurchase() is a good enough name

Reply:
Updated